### ISSUE TITLE
1.2.0: add plugin prepends all root configs; non-string values cause error

### ISSUE THREAD

**Issue by jrr** [dotansimha/graphql-code-generator#1896]
**Describe the bug**
1.2.0 does this:

```
$> gql-gen -o
  ✔ Parse configuration
  ❯ Generate outputs
    ❯ Generate modules/client/graphql/types.gen.tsx
      ✔ Load GraphQL schemas
      ✔ Load GraphQL documents
      ✖ Generate
        → a.startsWith is not a function
    ✔ Generate modules/client/graphql/client-types.gen.ts
    ✔ Generate modules/graphql-api/server-types.gen.ts


 Found 1 error

  ✖ modules/client/graphql/types.gen.tsx
    TypeError: a.startsWith is not a function


Something went wrong
```

I suspect it may have broken with this change https://github.com/dotansimha/graphql-code-generator/pull/1869, as it's the only instance of `a.startsWith` I could find in the repo.

I haven't gotten to the bottom of this yet, but here's a mostly-minimized repro config:

```yml
generates:
  modules/client/graphql/types.gen.tsx:
    config:
      strict: true
    schema:
      - "modules/graphql-api/schema.graphql"
      - "modules/client/graphql/schema.graphql"
    documents:
      - "modules/client/graphql/*/*.graphql"
    plugins:
      - add:
          - // hello
      - typescript
```

It looks like perhaps there's some interaction of `strict: true` and the _add_ plugin; remove either one and the error goes away.

**Comment by jrr** [comment]
Ah-ha! The _add_ plugin is picking up my root-level configs.

Given the above codegen.yml, and this code from the add plugin:

https://github.com/dotansimha/graphql-code-generator/blob/9a70bf6dcc8b7278308e1759dc5b58127288ad6a/packages/plugins/other/add/src/index.ts#L6-L13

A little bit of logging:

```
$> gql-gen -o
plugin { add: [ '// hello' ] }
config: { '0': '// hello', strict: true }
asArray: [ '// hello', true ]
prepending // hello
prepending true
plugin { typescript: {} }
prepending export type Maybe<T> = T | null;
a is string |// hello|
a is boolean |true|
  ✔ Parse configuration
  ❯ Generate outputs
    ❯ Generate modules/client/graphql/types.gen.tsx
      ✔ Load GraphQL schemas
      ✔ Load GraphQL documents
      ✖ Generate
        → a.startsWith is not a function


 Found 1 error

  ✖ modules/client/graphql/types.gen.tsx
    TypeError: a.startsWith is not a function


Something went wrong
```

Perhaps the plugin shouldn't look at root-level configs? Is that even possible?

**Comment by AnotherHermit** [comment]
I think the problem is that the config parameter that the add plugin receives is always either

a string:
```
...
- add: "some string"
```
or an object:
```
...
- add: 
  - "some string" (creates an object containing all the global config options)
- add:
  content: "string" (also an object)
```

So to solve your issue you can just add the string as:
```
...
- add: // hello
```

**Comment by dotansimha** [comment]
Available in 1.2.1
